lib/libc/common/source/time/clock.c

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2023 Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <time.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys_clock.h>
+
+#ifndef CLOCKS_PER_SEC
+#warning libc does not define CLOCKS_PER_SEC which is required by ISO
+#define CLOCKS_PER_SEC CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC
+#endif
+
+clock_t clock(void)
+{
+	return z_tmcvt(k_uptime_get(), MSEC_PER_SEC, CLOCKS_PER_SEC, true, sizeof(clock_t) <= 4,
```

Comment:
This should use `k_uptime_ticks` to provide the highest possible timer resolution, as well as to avoid double time conversions.